File: lib/naming-traps/legacy/frontend-services/UserService.ts
/**
 * DECOY — legacy frontend user service stub.
 * Production user logic lives in apps/api/meridian_api/services/user_service.py
 */

export interface User {
  id: string
  email: string
  name: string
}

export class UserService {
  async getUser(id: string): Promise<User | null> {
    console.warn('Legacy UserService stub')
    return null
  }

  async getUserByEmail(email: string): Promise<User | null> {
    return null
  }
}

File: apps/api/meridian_api/services/user_service.py
"""Canonical user service for the Meridian API.

When editing user lookup logic, THIS is the production file.
"""

from uuid import UUID

from psycopg2.extensions import connection

from meridian_api.errors import APIError


def get_user_by_id(db: connection, user_id: UUID) -> dict:
    with db.cursor() as cur:
        cur.execute(
            "SELECT id, email, name, created_at FROM users WHERE id = %s",
            (str(user_id),),
        )
        row = cur.fetchone()
    if not row:
        raise APIError("USER_NOT_FOUND", "User not found", 404)
    return dict(row)


def get_user_by_email(db: connection, email: str) -> dict | None:
    with db.cursor() as cur:
        cur.execute(
            "SELECT id, email, name, created_at FROM users WHERE email = %s",
            (email,),
        )
        row = cur.fetchone()
    return dict(row) if row else None